### DISCUSSION TITLE
Option to manually assign icons - icon gallery

### DISCUSSION THREAD

**Discussion by sv3nnie** [ente-io/ente#1036] [Enhancements]
Currently, icons are selected automatically based on the issuer name and there is no option to manually select an icon. However, I have encountered situations where I added a prefix to the issuer field in order to prioritize certain items or for other reasons. This resulted in those items not being assigned an icon automatically. It would be nice to have the ability to manually assign icons by selecting from an icon gallery or a similar feature that displays all available icons.

**Comment by JamsRepos** [comment]
I would say this solution would be perfect however, if this is sorta big task another option could be to be able to allow defining a URL as to where what site the 2FA is for? That way, it could be a little better than just having to name your 2FA codes what it's for. 

However, this would cause issues for self-hosted instances. For example, if I was to self-host my own vaultwarden and set the URL to https://passwords.domain.com/ there would be no way for it to be able to detect it. 

I hope that this solution is possible and comes into play though.

**Comment by RadCod3** [comment]
+1 
iirc Authy would grab the icon from the website’s favicon. That too maybe an option y'all could consider? Since I imagine storing images that users upload on your cloud wouldn’t be ideal.

Raivo had a separate icon repo that you could contribute icons to and the app would pull the icons from the repo. Meaning the app wouldn’t have to be rebuilt every time someone contributed a new icon. A similar approach could work for ente as well I imagine. It would be even better if said repo was customizable. As in I’d be able to fork it and add my own icons for my own services and point the app towards that without needing to merge it to the main repo, but that’s just a thought.

Really hope this gets implemented!

**Comment by daeh** [comment]
Authy seems to do this fairly well, and I assumed that they were doing fuzzy search on the item label (in addition to url matching). Currently, in Ente, if I label the item as "GitHub", the correct icon appears, but if I label it "GitHub: daeh" then no icon appears. I bet that implementing a more flexible way to parse the service name from the label would go a long way.

**Comment by ua741** [comment]
>  It would be nice to have the ability to manually assign icons by selecting from an icon gallery or a similar feature that displays all available icons.

This is now possible in v.4.2.0.
